addons/account/models/account_move_line.py

```diff
@@ -667,16 +667,26 @@ def _compute_amount_currency(self):
             if line.currency_id == line.company_id.currency_id:
                 line.amount_currency = line.balance
 
-    @api.depends('full_reconcile_id.name', 'matched_debit_ids', 'matched_credit_ids')
+    @api.depends('full_reconcile_id', 'matched_debit_ids', 'matched_credit_ids')
     def _compute_matching_number(self):
+        multi_partials = False
         for record in self:
             if record.full_reconcile_id:
-                record.matching_number = record.full_reconcile_id.name
+                record.matching_number = str(record.full_reconcile_id.id)
             elif record.matched_debit_ids or record.matched_credit_ids:
-                record.matching_number = 'P'
+                partials = record.matched_debit_ids or record.matched_credit_ids
+                if not multi_partials and len(partials) > 1:
+                    multi_partials = True
+                if len(partials) == 1:
+                    record.matching_number = 'P' + str(partials.id)
             else:
                 record.matching_number = None
 
+        if multi_partials:
+            for record in self:
+                if record.matched_debit_ids or record.matched_credit_ids:
+                    record.matching_number = 'P*'
```

Comment:
```py
@api.depends('full_reconcile_id', 'matched_debit_ids', 'matched_credit_ids')
def _compute_matching_number(self):
    for record in self:
        if record.full_reconcile_id:
            record.matching_number = set(record.full_reconcile_id.id)
        else:
            partials = record.matched_debit_ids or record.matched_credit_ids
            if len(partials) == 1:
                record.matching_number = f'P{partials.id}'
            elif partials:
                record.matching_number = 'P*'
            else:
                record.matching_number = None
```

Reply:
The result:
Transaction 1 ------
L1 Receivable Debit = P1
L2 Receivable Credit = P1
Transaction 2 -----
L1 Receivable Debit = P* (because there are multiple partials)
L3 Receivable Credit = P2 (because it only has one partial)

AFAIK @chklop wants L3 to be P* too?